base_multi_image/models/image.py

```diff
@@ -0,0 +1,239 @@
+    )
+    owner_model = fields.Char(
+        required=True,
+    )
+    owner_ref_id = fields.Reference(
+        selection="_selection_owner_ref_id",
+        string="Referenced Owner",
+        compute="_compute_owner_ref_id",
+        store=True,
+    )
+    storage = fields.Selection(
+        selection=[
+            ('url', 'URL'),
+            ('file', 'OS file'),
+            ('db', 'Database'),
+            ('filestore', 'Filestore'),
+        ],
+        required=True,
+    )
+    name = fields.Char(
+        string='Image title',
+        translate=True,
+    )
+    filename = fields.Char(
+        string='Filename',
+    )
+    extension = fields.Char(
+        string='File extension',
+        readonly=True,
+    )
+    attachment_id = fields.Many2one(
+        comodel_name='ir.attachment',
+        string='Attachment',
+        domain="[('index_content', '=', 'image')]",
+    )
+    file_db_store = fields.Binary(
+        string='Image stored in database',
+        filters='*.png,*.jpg,*.gif',
+    )
+    path = fields.Char(
+        string="Image path",
+        help="Image path",
+    )
+    url = fields.Char(
+        string='Image remote URL',
+    )
+    image_main = fields.Binary(
+        string="Full-sized image",
+        compute="_compute_image",
+    )
+    image_medium = fields.Binary(
+        "Medium-sized image",
+        compute="_compute_image_sizes",
+        help="Medium-sized image. It is automatically resized as a "
+             "128 x 128 px image, with aspect ratio preserved, only when the "
+             "image exceeds one of those sizes. Use this field in form views "
+             "or kanban views.")
+    image_small = fields.Binary(
+        "Small-sized image",
+        compute="_compute_image_sizes",
+        help="Small-sized image. It is automatically resized as a 64 x 64 px "
+             "image, with aspect ratio preserved. Use this field anywhere a "
+             "small image is required.")
+    comments = fields.Text(
+        'Comments',
+        translate=True)
+    sequence = fields.Integer(
+        default=10)
+    show_technical = fields.Boolean(
+        compute="_compute_technical")
+
+    @api.model
+    @tools.ormcache("self")
+    def _selection_owner_ref_id(self):
+        """Allow any model; after all, this field is readonly."""
+        return [(r.model, r.name) for r in self.env["ir.model"].search([])]
+
+    @api.multi
+    @api.depends("owner_model", "owner_id")
+    def _compute_owner_ref_id(self):
+        """Get a reference field based on the split model and id fields."""
+        for s in self:
+            s.owner_ref_id = "{0.owner_model},{0.owner_id}".format(s)
+
+    @api.multi
+    @api.depends('storage', 'path', 'file_db_store', 'url')
+    def _compute_image(self):
+        """Get image data from the right storage type."""
+        for s in self:
+            s.image_main = getattr(s, "_get_image_from_%s" % s.storage)()
```

Comment:
You changed the name of the function _get_image_from_filestore to _compute_image_from_filestore so don't work in that case.